BitwardenShared/UI/Platform/Settings/Extensions/Alert+Settings.swift

```diff
@@ -237,13 +237,27 @@ extension Alert {
 
     /// An alert asking if the user wants to login with their PIN upon app restart.
     ///
-    /// - Parameter action: The action to occur if `Yes` is tapped.
-    /// - Returns: An alert asking if the user wants to login with their PIN upon app restart.
+    /// - Parameters:
+    ///   - biometricType: The biometirc type the app supports.
+    ///   - action: The action to occur if `Yes` is tapped
     ///
-    static func unlockWithPINCodeAlert(action: @escaping (Bool) async -> Void) -> Alert {
-        Alert(
+    /// - Returns: An alert asking if the user wants to login with their PIN upon app restart.
+    static func unlockWithPINCodeAlert(
+        biometricType: BiometricAuthenticationType?,
+        action: @escaping (Bool) async -> Void
+    ) -> Alert {
+        var message = Localizations.pinRequireMasterPasswordRestart
+        if let biometricType {
+            switch biometricType {
+            case .faceID:
+                message = Localizations.pinRequireBioOrMasterPasswordRestart(Localizations.faceID)
+            case .touchID:
+                message = Localizations.pinRequireBioOrMasterPasswordRestart(Localizations.touchID)
+            }
+        }
```

Comment:
🎨 You could simplify this slightly by switching on the optional:
```suggestion
        let message = switch biometricType {
        case .faceID:
            Localizations.pinRequireBioOrMasterPasswordRestart(Localizations.faceID)
        case .touchID:
            Localizations.pinRequireBioOrMasterPasswordRestart(Localizations.touchID)
        case nil:
            Localizations.pinRequireMasterPasswordRestart
        }
```

Reply:
Maybe it's not much simpler, but reads a bit better? Your choice!

Reply:
Yeah much much better thanks! 🎉